addons/common/functions/fnc_nearestVehiclesFreeSeat.sqf

```diff
@@ -0,0 +1,21 @@
+/*
+ * Author: 654wak654
+ * Returns a list of vehicles near given unit that the unit can be a passenger in.
+ *
+ * Arguments:
+ * 0: The unit <OBJECT>
+ *
+ * Return Value:
+ * Nearest vehicles with a free seat <ARRAY>
+ *
+ * Example:
+ * [bob] call ace_common_fnc_nearestVehiclesFreeSeat
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_unit"];
+
+private _nearVehicles = nearestObjects [_unit, ["Car", "Air", "Tank", "Ship_F", "Pod_Heli_Transport_04_crewed_base_F"], 10];
```

Comment:
Making the distance configurable by allowing it to be passed along as a parameter would be more flexible.

```sqf
params ["_unit", ["_distance", 10]];

private _nearVehicles = nearestObjects [_unit, ["Car", "Air", "Tank", "Ship_F", "Pod_Heli_Transport_04_crewed_base_F"], _distance];
```

Reply:
Because of this function's very specific purpose it's good that everyone gets the same search distance IMO.

Reply:
It's a common function though. Name of it implies it finds nearest no matter the distance. As a common function I agree with @thojkooi, can always simply call it with distance of 10 from medical/captives.